Assets/MRTK/Core/Providers/Hands/BaseHand.cs

```diff
@@ -78,6 +101,99 @@ protected void UpdateVelocity()
             frameOn++;
         }
 
+        protected bool IsInTeleportPose
+        {
+            get
+            {
+                if (!TryGetJoint(TrackedHandJoint.Palm, out var palmPose)) return false;
+
+                Camera mainCamera = CameraCache.Main;
+
+                if (mainCamera == null)
+                {
+                    return false;
+                }
+
+                Transform cameraTransform = mainCamera.transform;
+
+                // We check if the palm up is roughly in line with the camera up
+                return Vector3.Dot(-palmPose.Up, cameraTransform.up) > 0.6f
+                       // Thumb must be extended, and middle must be grabbing
+                       && !isThumbGrabbing && isMiddleGrabbing;
+            }
+        }
+
+        // Used to track the input that was last raised
+        private Vector2 previousStickInput = Vector2.zero;
+        private bool previousReadyToTeleport = false;
+
+        private bool isIndexGrabbing;
+        private bool isMiddleGrabbing;
+        private bool isThumbGrabbing;
+
+        protected void UpdateTeleport()
+        {
+            MixedRealityInputAction teleportAction = MixedRealityInputAction.None;
+            IMixedRealityTeleportPointer teleportPointer = null;
+
+            // Check if we're focus locked or near something interactive to avoid teleporting unintentionally.
+            bool anyPointersLockedWithHand = false;
+            for (int i = 0; i < InputSource?.Pointers?.Length; i++)
+            {
+                if (InputSource.Pointers[i] == null) continue;
+                if (InputSource.Pointers[i] is IMixedRealityNearPointer)
+                {
+                    var nearPointer = (IMixedRealityNearPointer)InputSource.Pointers[i];
+                    anyPointersLockedWithHand |= nearPointer.IsNearObject;
+                }
+                anyPointersLockedWithHand |= InputSource.Pointers[i].IsFocusLocked;
+
+                // If official teleport mode and we have a teleport pointer registered, we get the input action to trigger it.
+                if (InputSource.Pointers[i] is IMixedRealityTeleportPointer)
+                {
+                    teleportPointer = (IMixedRealityTeleportPointer)InputSource.Pointers[i];
+                    teleportAction = teleportPointer.TeleportAction;
```

Comment:
Instead of hoping there's a teleport pointer present and asking it for its action, can this instead be set in a profile somewhere? This feels like a kinda circular action: "ask the teleport pointer for its action so i can send its action back to it". Instead, like the controller mappings, maybe we can set this in the hand tracking profile or maybe add it as a gesture mapping?

Reply:
Yeah this design pattern seemed a bit weird to me too. Where in the profile have we handled this similarly?